src/systems/user_commands/UserCommands.cc

```diff
@@ -953,6 +966,56 @@ bool UserCommandsPrivate::VisualService(const msgs::Visual &_req,
   return true;
 }
 
+//////////////////////////////////////////////////
+void UserCommandsPrivate::OnCmdMaterialColor(const msgs::MaterialColor &_msg)
+{
+  msgs::Visual _req;
+  int numberOfEntities = 0;
+  auto entities = entitiesFromScopedName(_msg.entity().name(),
+      *this->iface->ecm);
+  if (entities.empty())
+  {
+    gzwarn << "Entity name: " << _msg.entity().name() << ", is not found."
+           << std::endl;
+    return;
+  }
+  for (const Entity &id : entities)
+  {
+    if ((numberOfEntities > 0) && (_msg.entity_match() !=
+      gz::msgs::MaterialColor::EntityMatch::MaterialColor_EntityMatch_ALL))
+    {
+      break;
+    }
+    numberOfEntities++;
+    auto msg = _req.New();
+    msg->set_id(id);
```

Comment:
Are we skipping `shininess`?

Reply:
Okay. I would suggest leaving a TODO comment then.

Reply:
Makes sense, added.